src/extractor/guidance/turn_handler.cpp

```diff
@@ -476,108 +587,153 @@ std::pair<std::size_t, std::size_t> TurnHandler::findFork(const EdgeID via_edge,
+        }
+    }
+    // check all road classes and whether any road is obvious by its class
+    // @CHAU_TODO check whether this is not already covered by `has_obvious`
+    // @CHAU_TODO also check whether this makes sense
+    for (std::size_t base_index = right; base_index <= left; ++base_index)
+    {
+        const auto base_class =
+            node_based_graph.GetEdgeData(intersection[base_index].eid).road_classification;
+        for (std::size_t compare_index = right; compare_index <= left; ++compare_index)
+        {
+            if (base_index == compare_index)
             {
-                return isObviousOfTwo(via_edge, intersection[right + 1], intersection[right]) ||
-                       isObviousOfTwo(via_edge, intersection[right], intersection[right + 1]) ||
-                       isObviousOfTwo(via_edge, intersection[left], intersection[right + 1]) ||
-                       isObviousOfTwo(via_edge, intersection[right + 1], intersection[left]);
+                continue;
             }
-            return false;
-        }();
-
-        // A fork can only happen between edges of similar types where none of the ones is obvious
-        const bool has_compatible_classes = [&]() {
-            const bool ramp_class = node_based_graph.GetEdgeData(intersection[right].eid)
-                                        .road_classification.IsLinkClass();
-            for (std::size_t index = right + 1; index <= left; ++index)
-                if (ramp_class !=
-                    node_based_graph.GetEdgeData(intersection[index].eid)
-                        .road_classification.IsLinkClass())
-                    return false;
-
-            const auto in_classification =
-                node_based_graph.GetEdgeData(intersection[0].eid).road_classification;
-            for (std::size_t base_index = right; base_index <= left; ++base_index)
+            const auto compare_class =
+                node_based_graph.GetEdgeData(intersection[compare_index].eid).road_classification;
+            if (obviousByRoadClass(via_class, base_class, compare_class))
             {
-                const auto base_classification =
-                    node_based_graph.GetEdgeData(intersection[base_index].eid).road_classification;
-                for (std::size_t compare_index = right; compare_index <= left; ++compare_index)
-                {
-                    if (base_index == compare_index)
-                        continue;
-
-                    const auto compare_classification =
-                        node_based_graph.GetEdgeData(intersection[compare_index].eid)
-                            .road_classification;
-                    if (obviousByRoadClass(
-                            in_classification, base_classification, compare_classification))
-                        return false;
-                }
+                return false;
             }
-            return true;
-        }();
+        }
+    }
+    return true;
+}
```

Comment:
Can you express these two nested raw for loops via stdlib algorithms?
The outer one looks like a for each, the inner one like a any-of.